bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/LcnModuleHandler.java

```diff
@@ -169,12 +180,23 @@ public void handleCommand(ChannelUID channelUid, Command command) {
                 subHandler.handleCommandString((StringType) command, number.get());
             } else if (command instanceof DecimalType) {
                 DecimalType decimalType = (DecimalType) command;
-                DecimalType nativeValue = getConverter(channelUid).onCommandFromItem(decimalType.doubleValue());
-                subHandler.handleCommandDecimal(nativeValue, channelGroup, number.get());
+                Converter valueConverter = getConverter(channelUid);
+                if (valueConverter instanceof ValueConverter) {
+                    DecimalType nativeValue = ((ValueConverter) valueConverter)
+                            .onCommandFromItem(decimalType.doubleValue());
+                    subHandler.handleCommandDecimal(nativeValue, channelGroup, number.get());
+                }else{
```

Comment:
This casting bothers me a bit. Is there an other way?

Reply:
Sorry, had some issues with my eclipse/EGit installation. I hopefully managed to "clean" the branch from the commits not belonging to this PR, but I still have the following issues:

- I can't satisfy DCO (your commit apparently was missing signature). Tried to apply the steps mentioned in the DCO log but with no success.
- J-N-K is still listed as a reviewer for this PR, although the commits not belonging to this PR are now (hopefully) removed.

To your question regarding the cast: One could move `ValueConverter.onCommandFromItem(double)` and `ValueConverter.onCommandFromItem(QuantityType<?>)` up to `Converter` again making them `Nullable` so `InversionConverter` could return `null`. But in my opinion this would violate separation of concerns, see my last comment.

From a design perspective it might be better to separate conversion for representation (lcn bus -> UI) from conversion of commands (UI -> lcn bus) leading to two different converter hierarchies/interfaces, e.g. `StateRepresentationConverter.onStateUpdateFromHandler` and `CommandConverter.onCommandFromItem` - or to be okay with the cast ;-)

Reply:
I'd go for a modified version of your first proposal. Move the methods back to `Converter`. Make `Converter` a class and add a default implementation for all four methods, that return the identity. Then, Nullable is not needed. Casting, `instanceof` and semantic based null values violate OOP paradigms most often.

I don't think splitting the converters by the message direction doesn't solve the problem as the differentiation between values and binary states is still necessary.

Reply:
Converters refactored as you requested.